dff/pipeline/pipeline/actor.py

```diff
@@ -56,6 +57,79 @@ def error_handler(error_msgs: list, msg: str, exception: Optional[Exception] = N
         logger.error(msg, exc_info=exception)
 
 
+def types_match(type1: Type, type2: Type) -> bool:
+    """
+    This function compares types with assumption that one of the types might be a :py:class:`typing.ForwardRef`.
+    If it is so, it compares type and forward reference by name.
+
+    :param type1: First type to compare.
+    :param type2: Second type to compare.
+    :return: True if types are equal, False otherwise.
+    """
+    if type1 == type2:
+        return True
+    elif type(type1) is ForwardRef or type(type2) is ForwardRef:
+        type1_name = type1.__forward_arg__ if type(type1) is ForwardRef else type1.__name__
+        type2_name = type2.__forward_arg__ if type(type2) is ForwardRef else type2.__name__
+        return type1_name == type2_name
+    else:
+        return False
+
+
+def validate_callable(
+    callable: Callable,
+    name: str,
+    flow_label: str,
+    node_label: str,
+    error_msgs: list,
+    logging_flag: bool = True,
+    expected_types: Optional[List[Type]] = None,
+    return_type: Optional[Tuple[Type]] = None,
```

Comment:
Because `None` is also a type.
Otherwise, it would be impossible to distinguish cases when we:

1. Want to check return type, and the expected return type is `None`.
2. Do not want to check return type and thus passing `None` instead of the return type.

If we wrap return type into one-element `Tuple`, the difference becomes obvious: `None != (None,)`.